src/Adapter/Product/Repository/ProductRepository.php

```diff
@@ -96,27 +92,28 @@ class ProductRepository extends AbstractObjectModelRepository
      * @param Connection $connection
      * @param string $dbPrefix
      * @param ProductValidator $productValidator
-     * @param int $defaultCategoryId
      * @param TaxRulesGroupRepository $taxRulesGroupRepository
      * @param ManufacturerRepository $manufacturerRepository
      */
     public function __construct(
         Connection $connection,
         string $dbPrefix,
         ProductValidator $productValidator,
-        int $defaultCategoryId,
         TaxRulesGroupRepository $taxRulesGroupRepository,
         ManufacturerRepository $manufacturerRepository
     ) {
         $this->connection = $connection;
         $this->dbPrefix = $dbPrefix;
         $this->productValidator = $productValidator;
-        $this->defaultCategoryId = $defaultCategoryId;
         $this->taxRulesGroupRepository = $taxRulesGroupRepository;
         $this->manufacturerRepository = $manufacturerRepository;
     }
 
     /**
+     * @todo: Not sure this should be in the repository as it gives a false feeling the the repository can duplicate a
```

Comment:
What about we remove it then? Or throw an exception?

Reply:
Because it's not the scope of this PR at all, it's just to avoid forgetting this refacto is required